mantidproject/mantid
# Correct unit output in PEARL tof_xye files

**Description of work.**

When the SaveFocusedXYE output for tof_xye files was added to the PEARL scripts, [one of the save lines](https://github.com/mantidproject/mantid/commit/c42a9bb150c1062ce40663e6566115e3db4adb64#diff-f29b3ad1488000a6c8575504bcd9c59b3afb7fd0cd5ea50db2ed9d9163294a63R62-R66) appeared after a unit conversion from TOf to dSpacing. This simply moves it before, so the output x-axis data is correctly in TOF.

**To test:**
Download [PearlFocusBugFix-focusall.zip](https://github.com/mantidproject/mantid/files/5959928/PearlFocusBugFix-focusall.zip)
Extract 
Open the Test_TOF_Dsp.py script in Workbench and correct the file paths (including in the User_experiment.yaml file)
The script focuses some data, with`focus_mode="all"`.

Previously the output file `PRL101540_tt70_tof-0.xye` had x-axis values around 0.3, so was clearly in Angstroms. Now the x values are around 1500, so in microsec. You can check the exact values are the x-axis bin centres of the workspace `PRL101540_tt70_mods1-9_TOF_converted_group` in the TOF_converted_group

Fixes #30110



---

#### Reviewer ####

Please comment on the following ([full description](http://developer.mantidproject.org/ReviewingAPullRequest.html)):

##### Code Review #####

- Is the code of an acceptable quality?
- Does the code conform to the [coding standards](http://developer.mantidproject.org/Standards/)?
- Are the unit tests small and test the class in isolation?
- If there is GUI work does it follow the [GUI standards](http://developer.mantidproject.org/Standards/GUIStandards.html)?
- If there are changes in the release notes then do they describe the changes appropriately?

##### Functional Tests #####

- Do changes function as described? Add comments below that describe the tests performed?
- Do the changes handle unexpected situations, e.g. bad input?
- Has the relevant (user and developer) documentation been added/updated?

Does everything look good? Mark the review as **Approve**. A member of `@mantidproject/gatekeepers` will take care of it.
